.github/workflows/react-unit-test.yml

```
@@ -0,0 +1,31 @@
+name: React
+
+on:
+    push:
+      branches: [ "master" ]
+    pull_request:
+      branches: [ "master" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: gmt-web-app
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Check linting
+        run: npm run lint
+
+      - name: Format code
+        run: npm run format
```

Format modifies the code, here we can only run checks, so we need another script.

```suggestion
        run: npm run format:check
```